Fixing Energy Gun Weapon Description Runtimes + Miscellaneous Weapon Description Fixes, Take 2

SpaceDragon00 opened the pull request:
This is basically just #59372 except I forgot about that PR since I assumed it was merged

## About The Pull Request

Fixes a runtime issue cause by the possibility of loaded_projectile being null by changing how projectile stats are obtained on energy weapons, and adds a separate line for energy ammo types that deal stamina damage and regular damage to incorporate both damage types

## Why It's Good For The Game

Fewer runtimes is good

## Changelog
:cl:
fix: Secure safes no longer show up as weapons (I'm labelling this as a fix because I don't know any scenarios where you'd even be able to use a secure safe as an item, so its force value seemed unnecessary)
fix: PKAs and their derivatives (Ebows) now properly give info after the first shot
fix: Energy guns with no charge now properly give their info
fix: Energy guns that do stamina damage and regular damage now display a line in consideration of both the stamina and regular damage
fix: Conditional plurals in weapon description code have been fixed
code: Added a proc to initialize a list of object subtypes keyed to their object paths
code: Turned Hits_To_Crit into a define
code: Improved the clarity of some comments, along with adding some where they were previously missing
qol: A gun's weapon notes now gives you the notes of its magazine, which gives you the notes of its ammo, making it so you don't need to take out ammo to see its damage.
/:cl:

SpaceDragon00 committed:
Sing -> Singe Replacement

Self explanatory. Replaces sing with singe

SpaceDragon00 committed:
Removes Scorch, Just In Case

SpaceDragon00 committed:
Merge branch 'tgstation:master' into master

SpaceDragon00 committed:
Merge branch 'tgstation:master' into master

SpaceDragon00 committed:
Merge branch 'tgstation:master' into master

SpaceDragon00 committed:
Merge branch 'tgstation:master' into master

SpaceDragon00 committed:
Fixes: Secure Safe, KA, Energy

Creates a new version of the loaded projectile for energy weapons when analyzing its values. This fixes a number of issues, like not getting weapon info on KAs after the first shot, or not getting weapon info on energy weapons without charge

Removes the force value from secure safes. Why did it have one?

SpaceDragon00 committed:
Adding a Parentheses Pair Cause It Looks Nice

SpaceDragon00 committed:
Extra Safety If

I find it hard to believe there's a coded ammo casing without a projectile type but you can never be too sure

SpaceDragon00 committed:
Qdel Time

Adding qdel to make sure the created projectiles get collected

SpaceDragon00 committed:
Merge remote-tracking branch 'upstream/master' into wepdescfixes

SpaceDragon00 committed:
Resolving Merge Conflict, Stamina Damage, and Span Procs

- Added span procs
- Changed the combined stamina damage proc to only account for stamina since stamina damage does not stack with regular damage for the sake of crit

SpaceDragon00 committed:
Merge remote-tracking branch 'upstream/master' into wepdescfixes

Rohesie commented on code/modules/projectiles/guns/energy.dm:
```diff
@@ -65,11 +65,16 @@
-		exam_proj = for_ammo.loaded_projectile
+		if(!for_ammo.projectile_type)
+			continue
+		exam_proj = new for_ammo.projectile_type() // Initial loaded projectile cannot be used because its initial value is null
```
Make a global list of singletons per type instead.
Should be cleaner that creating and destroying one each time.

Rohesie commented on code/modules/projectiles/guns/energy.dm:
```diff
@@ -65,11 +65,16 @@
 	readout += "Standard models of this projectile weapon have [span_warning("[ammo_type.len]")] mode\s"
 	readout += "Our heroic interns have shown that one can theoretically stay standing after..."
 	for(var/obj/item/ammo_casing/energy/for_ammo in ammo_type)
```
If you could remove the type check here, using `as anything`, that'd be a code improvement.

Rohesie commented on code/game/objects/items/storage/secure.dm:
```diff
@@ -167,7 +167,6 @@
 	icon_locking = "safeb"
 	icon_sparking = "safespark"
 	desc = "Excellent for securing things away from grubby hands."
-	force = 8
```
Is this intended?

SpaceDragon00 replied:
People kept saying that the secure safe showing up as a weapon was a bug. I don't see how it could be used as a weapon so I didnt think it needed a force value, but I can revert it if need be

Rohesie replied:
Alright.

SpaceDragon00 commented:
Might stale due to delays from personal problems but I'll be working on the suggestions soon

SpaceDragon00 committed:
Global List, Span Procs, Clarity

- Added a global list of projectiles accessed by their path, rather than relying on loaded projectiles or creating new ones
- Added weapon notes for magazines, so you dont need to inspect ammo directly to get its notes
- Weapon notes now call other weapon notes for the sake of simplicity
- Improved the clarity of wording on comments and flavor text
- Fixed \s not working properly due to span procs

SpaceDragon00 committed:
Merge remote-tracking branch 'tgstation/master' into wepdescfixes

SpaceDragon00 committed:
Linters - Adds a null check to closets.dm

I dont know if this was a common linters problem, but since it's on a projectile I'll change it to make sure

SpaceDragon00 committed:
Linters Attempt 2

Let's try this again, this time with an early return
